### ISSUE TITLE
Custom UserService with custom repository

### ISSUE THREAD

**Issue by escar** [IdentityServer/IdentityServer3#2236]
Hi!

Based on the [EntityFramework](https://github.com/IdentityServer/IdentityServer3.Samples/tree/master/source/EntityFramework) and [CustomUserService](https://github.com/IdentityServer/IdentityServer3.Samples/tree/master/source/CustomUserService) samples, I'm trying to authenticate users against a separate user database.

In the factory, I'm registering the service like this:

    factory.UserService = new Registration<IUserService>(resolver => new CustomUserService());

However, if I want to use multiple user databases, how can I inject the repository or the connection string that I want to use? 

The goal is have different clients getting authorization to use resources (like API) from the same authorization server, but point to different user repository. So, identity server would use a database where clients, scopes, redirect URIs, etxc, would be set. But to authenticate a user, I would like to use different database, depending on the client.

Is there any way of doing this?

Thanks

**Comment by brockallen** [comment]
The client id is passed into the user service APIs in the context params.

**Comment by escar** [comment]
Thank you Brock.

So, I could extend Client schema and add a column that would contain the context to use. Is this a good approach?

**Comment by brockallen** [comment]
I don't think it's the best approach to extend IdentityServer's object model. But you could have your own mapping table with the info you need.

**Comment by escar** [comment]
Ok, thank you.

I'll try to figure out what to do...

**Comment by LinqBeast** [comment]
I started by looping through each user store and creating a single user model collection from them, which I then passed into my customer user service, first thinking the decorator cache implementation would magically know how to cache the users... that I could bust/add to later from each app when modified. Ha.  Muh bad.  I then considered caching my model collection and querying it upon authentication/profile lookup etc.. Ultimately, I settled with passing the tenant from the client and keying off a custom tenant claim to query the correct user store.
